static/js/reports/calibrations.js

```diff
@@ -0,0 +1,281 @@
+function report_calibrations(datastorage,daystoshow,options) {
+	var treatments = [];
+	Object.keys(daystoshow).forEach(function (day) {
+		treatments = treatments.concat(datastorage[day].treatments.filter(function (t) {
+			if (t.eventType == "Sensor Start") return true;
+			if (t.eventType == "Sensor Change") return true;
+			return false;
+		}));
+	});
+
+	var cals = [];
+	Object.keys(daystoshow).forEach(function (day) {
+		cals = cals.concat(datastorage[day].cal);
+	});
+
+	var sgvs = [];
+	Object.keys(daystoshow).forEach(function (day) {
+		sgvs = sgvs.concat(datastorage[day].sgv);
+	});
+		
+	var mbgs = [];
+	Object.keys(daystoshow).forEach(function (day) {
+		mbgs = mbgs.concat(datastorage[day].mbg);
+	});
+	mbgs.forEach(function (mbg) { calibrations_calcmbg(mbg); });	
+	
+
+	var events = treatments.concat(cals).concat(mbgs).sort(function(a, b) { return a.x - b.x; });
+	
+	var colors = ['Aqua','Blue','Brown','Chartreuse','Coral','CornflowerBlue','DarkCyan','DarkMagenta','DarkOrange','Fuchsia','Green','Yellow'];
+	var colorindex = 0;
+	var html = '<table>';
+	var lastmbg = null;
+	for (var i=0; i<events.length; i++) {
+		e = events[i];
+		var currentcolor = 'White';
+		if (typeof e.device !== 'undefined') colorindex = (colorindex+1)%colors.length;
+		if (!e.eventType) currentcolor = colors[colorindex];
+		html += '<tr>';
+		html += '<td>' + localeDateTime(new Date(e.x)) + '</td><td style="background-color:'+currentcolor+'">';
+		e.bgcolor = colors[colorindex];
+		if (e.eventType)
+			html += '<b style="text-decoration: underline;padding-left:0em">'+translate(e.eventType)+'</b>:<br>';
+		else if (typeof e.device !== 'undefined') {
+			html += '<input type="checkbox" index="'+i+'" class="calibrations-checkbox" id="calibrations-'+i+'"> ';
+			html += '<b style="padding-left:2em">MBG</b>: ' + e.y + ' Raw: '+e.raw+'<br>';
+			lastmbg = e;
+			e.cals = [];
+			e.checked = false;
+		} else if (typeof e.scale !== 'undefined') {
+			html += '<b style="padding-left:4em">CAL</b>: ' + ' Scale: ' + e.scale.toFixed(2) + ' Intercept: ' + e.intercept.toFixed(0) + ' Slope: ' + e.slope.toFixed(2) + '<br>';
+			if (lastmbg) lastmbg.cals.push(e);
+		} else html += JSON.stringify(e);
+		html += '</td></tr>';
+	};
+	
+	html += '</table>';
+
+	$('#calibrations-list').html(html);
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Mixed double and single quotes.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/769498832/issues/source?bid=2452374&fileBranchId=2459081#l59)